src/JsonSchema/Constraints/Constraint.php

```diff
@@ -83,19 +66,12 @@ public function getFactory()
      */
     public function getSchemaStorage()
     {
-        if (is_null($this->schemaStorage)) {
-            $this->schemaStorage = new SchemaStorage($this->getUriRetriever());
-        }
-
-        return $this->schemaStorage;
+        return $this->factory->getSchemaStorage();
     }
 
-    /**
-     * @param UriRetrieverInterface $uriRetriever
-     */
-    public function setUriRetriever(UriRetrieverInterface $uriRetriever)
+    public function getCheckMode()
     {
-        $this->uriRetriever = $uriRetriever;
+        return $this->factory->getCheckMode();
```

Comment:
this kind of API tells me that your factory is not really a factory (or not only that)